src/material_library.cpp

```diff
@@ -0,0 +1,352 @@
+    db = H5Fopen(filename.c_str(), H5F_ACC_RDWR, fapl);
+  }
+  // Check if root_path exist and what type it is
+  std::string root_path = "/material_library";
+
+  H5O_info_t object_info;
+  herr_t status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+
+  // Check if root_path exist and what type it is
+  status =
+      H5Oget_info_by_name(db, root_path.c_str(), &object_info, H5P_DEFAULT);
+  if (status == 0) {
+    // "/material_library" not a group: fail!
+    if (object_info.type != H5O_TYPE_GROUP) {
+      throw std::runtime_error(
+          "Non-group/non-dataset object /material_library already exists in "
+          "the file. Can't write the Material");
+    } else {  // "/material_library" is a group get its hid
+      matlib_grp_id = H5Gopen2(db, root_path.c_str(), H5P_DEFAULT);
+    }
+  } else {  // "/material_library" does not exist -> create it !
+    matlib_grp_id = H5Gcreate2(db, root_path.c_str(), H5P_DEFAULT, H5P_DEFAULT,
+                               H5P_DEFAULT);
+  }
+
+  // Group "/root_path/datapath" does not exist create it
+  std::string full_path = root_path + datapath;
+  if (!h5wrap::path_exists(db, full_path)) {
+    data_id = H5Gcreate2(db, full_path.c_str(), H5P_DEFAULT, H5P_DEFAULT,
+                         H5P_DEFAULT);
+  } else {
+    data_id = H5Gopen2(db, full_path.c_str(), H5P_DEFAULT);
+  }
+  std::string compath = "composition";
+  std::string nucpath = "nuclidelist";
+  write_hdf5_nucpath(data_id, nucpath);
+  std::vector<int> nuc_list;
+  nuc_list.assign(nuclist.begin(), nuclist.end());
+  for (auto mat : material_library) {
+    mat.second->write_hdf5_datapath(data_id, compath, -0.0,
+                                    DEFAULT_MAT_CHUNKSIZE, nuc_list);
+  }
+
+  H5Fflush(db, H5F_SCOPE_GLOBAL);
+  H5Gclose(data_id);
+  H5Gclose(matlib_grp_id);
+  H5Fclose(db);
+}
+
+void pyne::MaterialLibrary::write_hdf5_nucpath(hid_t db,
+                                               std::string nucpath) const {
+  //
+  // Read in nuclist if available, write it out if not
+  //
+  // bool nucpath_exists = h5wrap::path_exists(db, nucpath);
+  int nuc_size;
+  nuc_size = nuclist.size();
+  // Create the data if it doesn't exist
+  int nuc_data[nuc_size];
+  {
+    int n = 0;
+    for (auto nuc : nuclist) {
+      nuc_data[n] = nuc;
+      n++;
+    }
+  }
+  hsize_t nuc_dims[1];
+  nuc_dims[0] = nuc_size;
+  hid_t nuc_space = H5Screate_simple(1, nuc_dims, NULL);
+  hid_t nuc_set = H5Dcreate2(db, nucpath.c_str(), H5T_NATIVE_INT, nuc_space,
+                             H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+  H5Dwrite(nuc_set, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, nuc_data);
+
+  H5Fflush(db, H5F_SCOPE_GLOBAL);
+  H5Sclose(nuc_space);
+  H5Dclose(nuc_set);
+}
+
+void pyne::MaterialLibrary::append_to_nuclist(const pyne::Material& mat) {
+  pyne::comp_map mat_comp = mat.comp;
+  for (auto nuclide : mat_comp) {
+    nuclist.insert(nuclide.first);
+  }
+}
+
+int pyne::MaterialLibrary::get_length_of_table(
```

Comment:
Didn't see an answer to my question about passing the file pointer of HDF5 object.  The file is always already open when this is called.